tachyon/base/containers/adapters.h

```diff
@@ -206,6 +206,93 @@ class ZippedAdapter {
   U& u_;
 };
 
+// Internal adapter class for implementing tachyon::base::Chained.
+template <typename T>
+class ChainedAdapter {
+ public:
+  class Iterator {
+   public:
+    using ItType = decltype(std::begin(std::declval<T&>()));
+
+    using difference_type = std::ptrdiff_t;
+    using value_type = iter_value_t<ItType>;
+    using pointer = value_type*;
+    using reference = value_type&;
+    using iterator_category = std::forward_iterator_tag;
+
+    static Iterator begin(T& first, T& second) {
+      return Iterator(std::begin(first), std::end(first), std::begin(second),
+                      std::end(second));
+    }
+    static Iterator end(T& first, T& second) {
+      return Iterator(std::end(second), std::end(first), std::begin(second),
+                      std::end(second));
```

Comment:
```suggestion
      return Iterator(std::begin(first), std::end(first), std::begin(second),
                      std::end(second), false);
```

Reply:
Isn't this correct since it's supposed to be the end?

Reply:
When you iterate from `begin()` to `end()`, then `is_first` is changed to `false`. So I think `flase` is the right value for the `end()`.

Reply:
@chokobole Oh I meant the suggestion you've made to change to std::begin(first).

Reply:
Oh okay got it!